src/core/execute.c

```diff
@@ -2953,11 +2953,10 @@ static int exec_child(
                 }
         }
 
-        if (context->nice_set)
-                if (setpriority(PRIO_PROCESS, 0, context->nice) < 0) {
-                        *exit_status = EXIT_NICE;
-                        return log_unit_error_errno(unit, errno, "Failed to set up process scheduling priority (nice level): %m");
-                }
+        if (context->nice_set) {
+                if (setpriority_closest(context->nice) < 0)
+                        log_unit_debug_errno(unit, errno, "Failed to set up process scheduling priority (nice level): %m");
```

Comment:
`errno` is not the right error code to log here now. `setpriority_closest()` returns errors as return value after all, hence you need to do:

```
r = setpriority_closest(…);
if (r < 0)
        log_unit_debug(…);
```